src/util/MmapVectorImpl.h

```diff
@@ -185,31 +188,38 @@ void MmapVector<T>::push_back(const T& el) {
 
 // ________________________________________________________________
 template <class T>
-void MmapVector<T>::open(size_t size, const T& defaultValue, string filename) {
-  open(size, filename);
+void MmapVector<T>::open(size_t size, const T& defaultValue, string filename,
+                         AccessPattern pattern) {
+  open(size, filename, pattern);
+  advise(AccessPattern::Sequential);
   for (size_t i = 0; i < _size; ++i) {
     _ptr[i] = defaultValue;
   }
+  advise(_pattern);
```

Comment:
interesting, so you readvise when changing the access pattern

Reply:
Did this answer your question? advise is the private implementation detail (call to madvise) and setAccessPattern the global interface, that let's us for example first advise to use sequential for insertion and then switch to random for access etc...
Concerning your question about the effect: According to the manpages the "sequential" mode does "heavy prefetching" and also evicts pages soon after they are written. I did not yet measure it and think this heavily depends on the overall system load and the kernel implementation (there is a reason why it's called "advise")